src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -94,10 +146,10 @@
                 refEx = answer.refEx;
             }
 
+            this.enumCtx = new EnumCtx(homeCtx, answer, homeCtx.clnt);
             // Ensures that context won't get closed from underneath us
-            this.homeCtx = homeCtx;
-            homeCtx.incEnumCount();
-            enumClnt = homeCtx.clnt; // remember
+            this.enumCtx.homeCtx.incEnumCount();
+            this.cleanable = LDAP_CLEANER.register(this, enumCtx);
```

Comment:
Use  `enumCtx.getHomeCts()` above.
Use `this.enumCtx` in call to `register`.
For consistency.